examples/model_selection/plot_grid_search_stats.py

```diff
@@ -0,0 +1,527 @@
+# We can now inspect the results of our search, sorted by their
+# `mean_test_score`:
+
+results_df = pd.DataFrame(search.cv_results_)
+results_df = results_df.rename(
+    index={0: 'linear', 1: 'poly2', 2: 'poly3', 3: 'rbf'}
+    ).sort_values(by=['rank_test_score'])
+results_df[
+    ['params', 'rank_test_score', 'mean_test_score', 'std_test_score']
+    ]
+
+
+###############################################################################
+# We can see that the estimator using the `rbf` kernel performed better,
+# closely followed by `linear`. Both estimators with a `poly` kernel performed
+# worse, with the one using a two degree polynomial achieving a much lower
+# perfomance than all other models.
+#
+# The output of GridSearchCV does not provide information on the certainty of
+# the differences between the models. To evaluate this, we need to conduct a
+# statistical test. Specifically, to contrast the performance of two models we
+# should statistically compare their AUC scores. There are 100 observations
+# (AUC scores) for each model as we repreated 10 times a 10-fold
+# cross-validation.
+#
+# However, the scores of the models are not independent: we iteratively used
+# the same partitions of the data to evaluate them, increasing the correlation
+# between the performance of the models. This means that some partitions of the
+# data can make the distinction of the classes particularly easy or hard to
+# find for all models, and thus their scores will co-vary.
+#
+# Let's inspect this partition effect by plotting the performance of all models
+# in each fold, and calculating the correlation between models across folds:
+
+# create df of model scores ordered by perfomance
+model_scores = results_df.filter(like='split')
+
+# plot 30 examples of dependency between cv fold and AUC scores
+fig, ax = plt.subplots()
+sns.lineplot(
+    data=model_scores.transpose().iloc[:30],
+    dashes='', palette='Set1', marker='o', alpha=.5, ax=ax
+    )
+ax.set_xlabel("CV fold", size=12, labelpad=10)
+ax.set_ylabel("Model AUC", size=12)
+ax.tick_params(bottom=True, labelbottom=False)
+plt.show()
+
+# print correlation of AUC scores across folds
+print(f"Correlation of models:\n {model_scores.transpose().corr()}")
+
+###############################################################################
+# We can observe that the performance of the models highly depends on the fold.
+#
+# As a consequence, if we assume independence between observations we will be
+# underestimating the variance computed in our statistical tests, increasing
+# the number of false positive errors (i.e. detecting a significant difference
+# between models when such does no exist) [1]_.
+#
+# Several methods have been developed to correct for the variance in these
+# cases. In this example we will explore one of these methods implemented
+# using two different statistical approaches: Frequentist and Bayesian.
```

Comment:
You are right that the frequentist does not need capital letters so I changed it in the example. The Bayesian apparently does need the capitalization since its origins are a surname, so all papers/tutorials I have read use it like this.